src/utilities/general.c

```diff
@@ -236,35 +269,19 @@ hypre_GetDeviceLastError(void)
  *****************************************************************************/
 
 HYPRE_Int
-HYPRE_Initialize(void)
+HYPRE_DeviceInitialize(void)
 {
-   /* Return if the hypre library is in initialized state */
-   if (hypre_Initialized())
-   {
-      return hypre_error_flag;
-   }
-
-#if defined(HYPRE_USING_MEMORY_TRACKER)
-   if (!_hypre_memory_tracker)
-   {
-      _hypre_memory_tracker = hypre_MemoryTrackerCreate();
-   }
-#endif
-
-   if (!_hypre_handle)
-   {
-      _hypre_handle = hypre_HandleCreate();
-   }
+   hypre_Handle *handle = hypre_handle();
 
 #if defined(HYPRE_USING_GPU) || defined(HYPRE_USING_DEVICE_OPENMP)
 #if defined(HYPRE_USING_SYCL)
    /* WM: note that for sycl, we need to do device setup again if reinitializing */
-   if (!HYPRE_Initialized())
+   //if (!HYPRE_Initialized())
```

Comment:
- [ ] Need to talk about this logics ....

Reply:
In my opinion, these lines can be removed:
https://github.com/hypre-space/hypre/blob/e6785d07e78485f6f5b6b79b5fe6a0dd50efcf88/src/utilities/general.c#L277-L285

Reply:
@victorapm K, let's talk about it. I want to confirm what we want to support for Init/ReInit, etc. Thanks